tests/vectors_kem.c

```diff
@@ -11,15 +11,34 @@
 #include <sys/stat.h>
 
 #include <oqs/oqs.h>
-
+#include <oqs/sha3.h>
 #include "system_info.c"
 
+#ifdef OQS_ENABLE_KEM_ML_KEM
+/* macros for sanity checks for encaps and decaps key */
+#define ML_KEM_BLOCKSIZE        384
+#define ML_KEM_K_MAX            4
+#define ML_KEM_N                256
+#define ML_KEM_1024_PK_SIZE     1568
+#define ML_KEM_Q                3329
+#define SHA256_OP_LEN           32
+/* since x is 12 bits, max value could be 4095. the below macro uses this to implement a simple time constant mod 3329 */
+#define MOD_Q(x)             ((x) - ((x >= ML_KEM_Q) * ML_KEM_Q))
```

Comment:
I don't believe we can count on comparison operators being constant time. I suggest doing Barrett reduction here instead, similar to the [reference implementation](https://github.com/open-quantum-safe/liboqs/blob/dd1706bc9bcd13dc4c909dcae057227449e9a14a/src/kem/ml_kem/pqcrystals-kyber-standard_ml-kem-768_ref/reduce.c#L35). (That code computes a centred representation; we'd just need an additional addition.)

I realize that this is overkill for testing code, but there's a possibility we use this file as a guide for patching the algorithm source later.

Reply:
correct, added a simple mod function as this was a test file. will change it to proper mod function after few tests at my end.

Reply:
Hi @SWilson4 , thanks for the update.
But any particular reason, why non-constant time is ok ? having it time constant would be the ideal scenario IMO.
in the meantime, I have tested modQ using "Barrett reduction" and another approach using shift operators as well.

Reply:
> Hi @SWilson4 , thanks for the update. But any particular reason, why non-constant time is ok ? having it time constant would be the ideal scenario IMO. in the meantime, I have tested modQ using "Barrett reduction" and another approach using shift operators as well.

I brought it up with the OQS team, and we decided that we're OK with a non–constant time function here because the code is limited to a test file, in the interest of not holding up the PR. If you prefer to submit a constant-time implementation, that's fine too :)